tests/candev/main.c

```diff
@@ -192,10 +206,16 @@ int main(void)
     puts("candev test application\n");
 
     isrpipe_init(&rxbuf, (uint8_t *)rx_ringbuf, sizeof(rx_ringbuf));
-#ifdef BOARD_NATIVE
+#if defined (BOARD_NATIVE)
     puts("Initializing Linux Can device");
     candev_linux_init( &linux_dev, &(candev_linux_conf[0]));    /* vcan0 */
     candev = (candev_t *)&linux_dev;
+
+#elif  defined(CAN_DRIVER_MCP2515)
+    puts("Initializing MCP2515");
+    candev_mcp2515_init(&mcp2515_dev, &mcp2515_conf);
```

Comment:
I'm a little confused. Shouldn't this take place already in `auto_init`? Surely, I see that you just took over the logic from native which confuses me as well...

Reply:
No, because the whole point of using this candev app is to use the candev abstraction layer without the can-DLL. 
We made `auto_init_can` explicit in an earlier PR. It should not be enabled here, and will most likely cause some issues if it does get enabled.

What exactly do you mean with "You took over the logic from native, which confuses me as well..." ?

Reply:
> No, because the whole point of using this candev app is to use the candev abstraction layer without the can-DLL.

I still don't really understand it. But this is out of scope for the PR anyway. @JannesVolkens can you have an eye on it?

 > We made auto_init_can explicit in an earlier PR. It should not be enabled here, and will most likely cause some issues if it does get enabled.

If I'm not mistaken, `auto_init` is always enabled and needs do be disabled explicitly. This is not the case in the applications Makefile.

> What exactly do you mean with "You took over the logic from native, which confuses me as well..." ?

You did the same as `candev_linux_init` does which is ok for the moment.

Reply:
> If I'm not mistaken, auto_init is always enabled and needs do be disabled explicitly. This is not the case in the applications Makefile.

This commit might clear up the confusion:
https://github.com/RIOT-OS/RIOT/pull/13360/commits/9b829c99a98f5b09c644c203b36f1462464f77ee